script/arrays.js

```diff
@@ -0,0 +1,221 @@
+shopingList[2] = new Product("orange", 5, false, 10);
+shopingList[3] = new Product("sausage", 1, false, 100);
+shopingList[4] = new Product("bread", 1, false, 27);
+shopingList[5] = new Product("lemon", 2, true, 7);
+shopingList[6] = new Product("avocado", 1, false, 55);
+
+//  Виводити весь список на екран таким чином, щоб спочатку йшли продукти, що ще не придбані, а потім - ті, що вже придбали.
+
+shopListOrdered = (arr) => {
+    console.log("Shopping List");
+    for (let i = 0; i < arr.length; i++) {
+        if (arr[i].bought == true) {
+            continue;
+        }
+        console.log(arr[i]);
+    }
+    for (let i = 0; i < arr.length; i++) {
+        if (arr[i].bought == true) {
+            console.log(arr[i]);
+        }
+    }
+}
+
+shopListOrdered(shopingList);
+
+//  Покупка продукту. Функція приймає назву продукту і відзначає його як придбаний.
+
+productPurchase = (productName) => {
+    for (let i = 0; i < shopingList.length; i++) {
+        if (shopingList[i].name == productName && shopingList[i].bought == true) {
+            console.log(`You have already bought ${productName}.`);
+            break;
+        }
+        if (shopingList[i].name == productName && shopingList[i].bought == false) {
+            shopingList[i].bought = true;
+            console.log(`You have successfully purchased ${productName}.`);
+            break;
+        } else {
+            console.log(`${productName} is not in the shopping list.`);
+        }
+    }
+}
+
+//  Створення списку (не) придбаних продуктів.
+
+shopListUnpurchased = (arr) => {
+    console.log("Shopping List of Unpurchased Products");
+    for (let i = 0; i < arr.length; i++) {
+        if (arr[i].bought == false) {
+            console.log(arr[i]); 
+            // console.log(arr[i].name); - якщо треба тільки назви продуктів вивести
+        } 
+    }
+}
+
+productPurchase("apple");
+shopListUnpurchased(shopingList);
+
+
+// standart
+
+
+// 1. Видалення продукту зі списку (видалення повинно проводитися шляхом створення нового масиву, 
+// в якому продукт, що ми шукаємо, буде відсутнім)
+
+productRemove = (productName) => {
+    shopingList = shopingList.map((item) => {
+        if (item.name != productName) {
+            return item;
+        }
+    });
+
+    for (let i = 0; i < shopingList.length; i++) {
+        if (shopingList[i] == undefined) {
+            shopingList.splice(i, 1);
+        } 
+    }
+    return shopingList;
+}
+
+console.log(shopingList);
+productRemove("apple");
+console.log(shopingList);
+productRemove("cheese");
+console.log(shopingList);
+
+
+// 2. Додавання покупки в список. Враховуй, що при додаванні покупки з уже існуючим в списку продуктом, 
+// необхідно збільшувати кількість в існуючій покупці, а не додавати нову. При цьому також повинна змінитися сума, 
+// наприклад, якщо ціна за одиницю 12, а кількості товарів стало 2, то сума буде 24.
+
+function isPositiveNumber (a) {
+    if (a > 0 && a == +a) {
```

Comment:
Would be better to use ternary --- gold rule - less code, less bits, less mistakes 😄